include/ardpolyfills/Servo.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Servo.h
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef SMARTCAR_EMUL_SERVO_H
+#define SMARTCAR_EMUL_SERVO_H
+
+#include <cstdint>
+
+#define MIN_PULSE_WIDTH      1000
+#define MAX_PULSE_WIDTH      2000
+#define DEFAULT_PULSE_WIDTH  1500
+
+#define INVALID_SERVO         255
+
+class Servo
+{
+  public:
+    Servo();
+    uint8_t attach(int pin);
+    uint8_t attach(int pin, int min, int max);
+    void detach();
+    void write(int value);
+    void writeMicroseconds(int value);
+    int read();
+    int readMicroseconds();
+    bool attached();
```

Comment:
Should be `const noexcept`

Reply:
Header was probably taken from the official [implementation](https://github.com/arduino-libraries/Servo/blob/master/src/Servo.h), while noexcept is fine I think we should keep it non const.

Reply:
Why do you think it should be non-const?
`attached` should DEFINITELY not mutate any sort of state; I'd even go as far as saying it should be marked as `[[gnu::const]]` in the implementation

Reply:
> Header was probably taken from the official [implementation](https://github.com/arduino-libraries/Servo/blob/master/src/Servo.h)

You mean literally copy-pasted from the [AVR, SAM, SAMD, NRF52, STM32F4] implementation without giving any reasoning as to what can be made better in that piece of legacy code

Reply:
Like I said, to match the official implementation.
But constness shouldn't matter anyways so I wont block it

Reply:
We only refer to [the reference](https://www.arduino.cc/en/Reference/Servo)
Libraries implementations are to be expected to be implemented differently for different platforms
For the core Arduino library we made sure to match `virtual`-ness, because people actually rely on it

Reply:
Only for attached?

Reply:
Yes; the other functions are mutating state, either local or global